### PR TITLE
gh-363 request param to select finalized models by label

### PR DESCRIPTION
**Pull request by GBishop-PHSA**
gh-363 Provide a path in the API to view the latest finalised datamodel

eg. http://localhost:8080/api/dataModels/path/dm:DM_LABEL?finalised=true

### CODE REVIEW

**Comment by joe-crawford** [comment]
@GBishop-PHSA, thanks, this looks like a good solution, however there's some errors in the tests relating to the path endpoints for folders. I think this may need some more handling for Versioned Folders (which should work like DataModels).

https://jenkins.cs.ox.ac.uk/job/Mauro%20Data%20Mapper/job/mdm-core/job/PR-364/1/display/redirect

File: mdm-core/grails-app/controllers/uk/ac/ox/softeng/maurodatamapper/core/path/PathController.groovy [hunk 1]
```diff
@@ -58,7 +58,8 @@ class PathController extends RestfulController<CatalogueItem> implements MdmCont
             // Permissions have been checked as part of the interceptor
             pathedResource = pathService.findResourceByPathFromRootResource(resource as MdmDomain, params.path)
```
Comment (joe-crawford) [inline mdm-core/grails-app/controllers/uk/ac/ox/softeng/maurodatamapper/core/path/PathController.groovy:59]: Does this method `pathService.findResourceByPathFromRootResource` also need changing as it looks like a similar route into looking up a domain by its path?

**Review by joe-crawford — CHANGES_REQUESTED** [review]

**Comment by GBishop-PHSA** [comment]
@joe-crawford I'm having an issue getting the tests to fail locally so I'm not sure how to investigate the failures

./gradlew -Dgradle.functionalTest=true :mdm-core:integrationTest
3 failed tests in AsyncJobFunctionalSpec

./gradlew -Dgradle.functionalTest=true :mdm-plugin-datamodel:integrationTest
1 failed test in DataModelFunctionalSpec

**Comment by joe-crawford** [comment]
Thanks @GBishop-PHSA, this looks good and I'll merge it. I've made some minor tweaks in a branch `feature/gh-363` on this repo, could you pull those to your branch then I'll merge the PR?
- Auto reformat using the IntelliJ code style XML file
- Add extra abstract method without the params arg

The test failures originally were all HibernateOptimisticLockingFailureException or StaleStateException, basically these are sometimes expected failures of the Hibernate optimistic locking strategy that in actual use would need to be retried. They're related to a performance/reliability trade off of Hibernate, if they occurred more often in actual use it would be a problem but usually these happen just on the test server. They've passed after rerunning Jenkins.

**Comment by GBishop-PHSA** [comment]
@joe-crawford, Thanks for looking at this. I've pulled in your changes and squashed the commits.

**Comment by joe-crawford** [comment]
@GBishop-PHSA, Great, thank you!

**Review by joe-crawford — APPROVED** [review]

**Pull request merged**